src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/QueueItem.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using CommonLanguageServerProtocol.Framework;
+using Microsoft.VisualStudio.Threading;
+
+namespace CommonLanguageServerProtocol.Framework;
+
+public class QueueItem<TRequestType, TResponseType, RequestContextType> : IQueueItem<RequestContextType>
```

Comment:
As mentioned above, does this have to be public? Feels very much like an implementation detail of `RequestExecutionQueue`.. given that the queue can be entirely swapped out if consumers want to customize its behaviour, I think it makes more sense to keep this all internal. If we find a need for someone to customize something about the queue item, without wanting to create an entirely custom queue, we can probably address that later, and in a way that better suits the scenario (eg, is it an option they can set on the queue? a new bit of metadata on the handler? etc.)

Reply:
Fair duce, will make internal.